stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/utils/extensions/String.kt

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+package io.getstream.chat.android.ui.utils.extensions
+
+internal fun String.containsLinks(): Boolean {
+    val regex = """(?:\s|^)((?:https?:)?(?:[a-z0-9-]+(?:\.[a-z0-9-]+)+)(?::[0-9]+)?(?:(?:[^\s]+)?)?)""".toRegex()
```

Comment:
Can we please update the regex here to fix the capitalization issue? Also - can we pull this function out to `ui-common`, mark is as public, InternalStreamChatApi and then reuse in both Compose and XML?

That way we don't have to update it in two places for no reason.

Reply:
I was initially avoiding this because then it will still be visible to Kotlin users since we have to mark it as public.
Yes they have to opt in to actually use it but it can still be seen as autocomplete clutter.

Reply:
Resolved in a meeting.

Decided to place it in the common module.